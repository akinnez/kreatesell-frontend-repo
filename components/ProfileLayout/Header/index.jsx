import { useEffect, useState } from "react";
import Link from "next/link";
import { useSelector } from "react-redux";
import { Button, Dropdown, Layout, Menu } from "antd";
import { Logout } from "../../../redux/actions";
import { ProfileIcon, Cog, Bell } from "../../IconPack";
import Logo from "../Logo";
import { NotificationDropdown } from "components/notification/Dropdown";
import style from "./index.module.scss";

const Profile = ({ name }) => {
  return (
    <>
      <div className="profile-wrapper">
        <div id="profile-content">
          <h4>{name || "Undefined"}</h4>
          <p>Account</p>
        </div>
        <div className="profile">
          <ProfileIcon />
        </div>
      </div>
      <style jsx>{`
        .profile-wrapper {
          display: flex;
          align-items: center;
          gap: 10px;
        }

        #profile-content h4,
        #profile-content p {
          margin: 0;
          line-height: 1.1;
          text-align: right;
        }

        #profile-content h4 {
          color: rgba(89, 89, 89, 1);
          font-weight: bold;
        }

        #profile-content p {
          color: rgba(140, 140, 140, 1);
        }

        .profile {
          box-shadow: 0px 20px 27px rgba(0, 0, 0, 0.05);
          border-radius: 8px;
          width: 42px;
          height: 42px;
          background-color: #ffffff;
          display: flex;
          align-items: center;
          justify-content: center;
        }
      `}</style>
    </>
  );
};

const menu = logout => (
  <Menu>
    <Menu.Item key="prof-1">
      <Link href="/account/kreator/store/edit">
        <a>Profile</a>
      </Link>
    </Menu.Item>
    <Menu.Divider />
    <Menu.Item key="prof-2" onClick={() => logout()}>
      Logout
    </Menu.Item>
  </Menu>
);

const Nav = () => {
  const { Header } = Layout;

  const [info, setInfo] = useState({});
  const [showNotification, setShowNotification] = useState(false);

  const { notifications } = useSelector(state => state.notification);

  const unreadNotification = notifications?.filter(item => !item?.is_read);

  const logout = Logout();

  useEffect(() => {
    const user = JSON.parse(localStorage.getItem("user"));
    setInfo(user);
  }, []);

  return (
    <>
      <Header className={style.header}>
        <Logo />
        <div className={style.nav_right}>
          <Button type="text" shape="circle" icon={<Cog />} />
          <Button
            type="text"
            shape="circle"
            icon={<Bell />}
            onClick={() => setShowNotification(value => !value)}
          >
            {unreadNotification?.length > 0 && (
              <span className="red bg-red-500 absolute rounded-full h-5 w-5 text-white text-xs flex items-center justify-center -top-2 right-3">
                {unreadNotification.length}
              </span>
            )}
          </Button>
          <Dropdown overlay={menu(logout)} placement="bottomCenter" arrow>
            <Button type="text">
              <Profile name={info?.full_name} />
            </Button>
          </Dropdown>
        </div>
      </Header>
      {showNotification && <NotificationDropdown />}
    </>
  );
};

export default Nav;
