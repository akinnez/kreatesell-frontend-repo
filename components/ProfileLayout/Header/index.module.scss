@use 'public/css/mixins';
@use 'public/css/screens';

.header {
	@include mixins.flex(row, space-between, center);
	background-color: #fff;
	padding: 25px 10px;
	height: auto;
}

.edit {
	@include mixins.flex(row, null, center);
	gap: 10px !important;
}

.nav_right {
	@include mixins.flex(row, flex-end, center);
	column-gap: 35px;

	> button {
		justify-content: center;
		height: auto;

		&:first-child {
			display: none !important;
		}
	}
}

.dropdown__btn {
	padding: 0;
}

.authMenu {
	max-width: 200px !important;
	// border: 1px solid red;
	margin-left: auto;
}
@include screens.minScreen('tablet') {
	.header {
		padding: 25px;
	}

	.nav_right button {
		&:first-of-type {
			display: flex !important;
		}
	}
}

.lg {
	@media (max-width: 992px) {
		display: none;
	}
}
