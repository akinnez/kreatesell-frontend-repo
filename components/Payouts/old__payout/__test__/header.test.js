import renderer from "react-test-renderer";
import Header from "../header";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

describe("Header : ", () => {
  it("renders a snapshot of <Header />", () => {
    const header = renderer.create(<Header />).toJSON();
    expect(header).toMatchSnapshot();
  });
});
