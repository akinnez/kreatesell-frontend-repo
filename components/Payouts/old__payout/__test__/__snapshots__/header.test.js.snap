// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header :  renders a snapshot of <Header /> 1`] = `
Array [
  <div
    className="header_container"
  >
    <h3>
      Payouts
    </h3>
    <button
      className="ant-btn ant-btn-primary btn"
      onClick={[Function]}
      style={
        Object {
          "width": "180px",
        }
      }
      type="button"
    >
      <span>
        Payout Setting
      </span>
    </button>
  </div>,
  <form
    className="ant-form ant-form-vertical"
    onReset={[Function]}
    onSubmit={[Function]}
    style={
      Object {
        "marginTop": "20px",
        "paddingLeft": "25px",
      }
    }
  >
    <div
      className="ant-space ant-space-horizontal ant-space-align-center"
      style={
        Object {
          "flexWrap": "wrap",
          "marginBottom": -30,
        }
      }
    >
      <div
        className="ant-space-item"
        style={
          Object {
            "marginRight": 30,
            "paddingBottom": 30,
          }
        }
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="Search"
            >
              Search
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <input
                  className="ant-input ant-input-lg input"
                  extraLabel="Search"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Search"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-space-item"
        style={
          Object {
            "marginRight": 30,
            "paddingBottom": 30,
          }
        }
      >
        <div
          className="ant-row ant-form-item"
          defaultValue=""
          style={
            Object {
              "width": "200px",
            }
          }
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title=""
            >
              <label
                className="label"
              >
                Currency
                 
                <span />
              </label>
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <div
                  className="ant-select ant-select-lg input ant-select-single ant-select-show-arrow ant-select-show-search"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                >
                  <div
                    className="ant-select-selector"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                  >
                    <span
                      className="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autoComplete="off"
                        className="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        onChange={[Function]}
                        onCompositionEnd={[Function]}
                        onCompositionStart={[Function]}
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onPaste={[Function]}
                        readOnly={false}
                        role="combobox"
                        style={
                          Object {
                            "opacity": null,
                          }
                        }
                        type="search"
                        unselectable={null}
                        value=""
                      />
                    </span>
                    <span
                      className="ant-select-selection-placeholder"
                    >
                      NGN
                    </span>
                  </div>
                  <span
                    aria-hidden={true}
                    className="ant-select-arrow"
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "WebkitUserSelect": "none",
                        "userSelect": "none",
                      }
                    }
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      className="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-space-item"
        style={
          Object {
            "marginRight": 30,
            "paddingBottom": 30,
          }
        }
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="Show from"
            >
              Show from
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <div
                  className="ant-picker ant-picker-large input"
                  onMouseUp={[Function]}
                >
                  <div
                    className="ant-picker-input"
                  >
                    <input
                      autoComplete="off"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      placeholder="Select date"
                      readOnly={true}
                      size={12}
                      title=""
                      value=""
                    />
                    <span
                      className="ant-picker-suffix"
                    >
                      <span
                        aria-label="calendar"
                        className="anticon anticon-calendar"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="calendar"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-space-item"
        style={
          Object {
            "marginRight": 30,
            "paddingBottom": 30,
          }
        }
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="to"
            >
              to
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <div
                  className="ant-picker ant-picker-large input"
                  onMouseUp={[Function]}
                >
                  <div
                    className="ant-picker-input"
                  >
                    <input
                      autoComplete="off"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      placeholder="Select date"
                      readOnly={true}
                      size={12}
                      title=""
                      value=""
                    />
                    <span
                      className="ant-picker-suffix"
                    >
                      <span
                        aria-label="calendar"
                        className="anticon anticon-calendar"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="calendar"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-space-item"
        style={
          Object {
            "paddingBottom": 30,
          }
        }
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title=" "
            >
               
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <button
                  className="ant-btn ant-btn-primary btn"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 25 24"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.4999 7.5C12.4999 8.95869 13.0793 10.3576 14.1108 11.3891C15.1422 12.4205 16.5412 13 17.9999 13C19.4586 13 20.8575 12.4205 21.889 11.3891C22.9204 10.3576 23.4999 8.95869 23.4999 7.5C23.4999 6.04131 22.9204 4.64236 21.889 3.61091C20.8575 2.57946 19.4586 2 17.9999 2C16.5412 2 15.1422 2.57946 14.1108 3.61091C13.0793 4.64236 12.4999 6.04131 12.4999 7.5ZM20.9999 4C21.1325 4 21.2597 4.05268 21.3534 4.14645C21.4472 4.24021 21.4999 4.36739 21.4999 4.5V6.5C21.4999 6.63261 21.4472 6.75979 21.3534 6.85355C21.2597 6.94732 21.1325 7 20.9999 7H18.9999C18.8673 7 18.7401 6.94732 18.6463 6.85355C18.5525 6.75979 18.4999 6.63261 18.4999 6.5C18.4999 6.36739 18.5525 6.24021 18.6463 6.14645C18.7401 6.05268 18.8673 6 18.9999 6H19.9999C19.7672 5.68928 19.4653 5.4371 19.1181 5.26351C18.7709 5.08992 18.388 4.99969 17.9999 5C17.1669 5 16.4279 5.407 15.9729 6.036C15.8939 6.14012 15.7774 6.2092 15.6481 6.22845C15.5189 6.2477 15.3872 6.2156 15.2814 6.13901C15.1755 6.06242 15.1038 5.94743 15.0817 5.81866C15.0595 5.68988 15.0887 5.55756 15.1629 5.45C15.4871 5.00083 15.9133 4.63513 16.4066 4.38303C16.8998 4.13093 17.4459 3.99965 17.9999 4C18.9799 4 19.8649 4.403 20.4999 5.05V4.5C20.4999 4.36739 20.5525 4.24021 20.6463 4.14645C20.7401 4.05268 20.8673 4 20.9999 4ZM15.4999 9.95V10.5C15.4999 10.6326 15.4472 10.7598 15.3534 10.8536C15.2597 10.9473 15.1325 11 14.9999 11C14.8673 11 14.7401 10.9473 14.6463 10.8536C14.5525 10.7598 14.4999 10.6326 14.4999 10.5V8.5C14.4999 8.36739 14.5525 8.24021 14.6463 8.14645C14.7401 8.05268 14.8673 8 14.9999 8H16.9999C17.1325 8 17.2597 8.05268 17.3534 8.14645C17.4472 8.24021 17.4999 8.36739 17.4999 8.5C17.4999 8.63261 17.4472 8.75979 17.3534 8.85355C17.2597 8.94732 17.1325 9 16.9999 9H15.9999C16.4559 9.608 17.1829 10 17.9999 10C18.7659 10 19.4519 9.656 19.9109 9.112C19.9528 9.06058 20.0046 9.01803 20.0632 8.98683C20.1217 8.95563 20.1859 8.9364 20.252 8.93026C20.3181 8.92412 20.3847 8.9312 20.4481 8.95109C20.5114 8.97097 20.5701 9.00325 20.6208 9.04607C20.6715 9.08888 20.7132 9.14135 20.7434 9.20044C20.7736 9.25952 20.7918 9.32403 20.7968 9.3902C20.8018 9.45638 20.7937 9.52289 20.7727 9.58586C20.7518 9.64884 20.7185 9.70701 20.6749 9.757C20.3466 10.1468 19.9371 10.46 19.475 10.6747C19.0129 10.8894 18.5094 11.0005 17.9999 11C17.5342 11.0006 17.0731 10.908 16.6437 10.7277C16.2144 10.5474 15.8254 10.283 15.4999 9.95ZM8.49987 13H14.5339C13.7248 12.4887 13.038 11.806 12.5219 11H8.49987L8.38287 11.007C8.12977 11.0371 7.89772 11.1627 7.73413 11.3582C7.57054 11.5536 7.48776 11.8042 7.5027 12.0586C7.51763 12.313 7.62916 12.5522 7.8145 12.7272C7.99984 12.9021 8.24499 12.9997 8.49987 13ZM5.49987 6H11.6739C11.5193 6.65468 11.4671 7.32933 11.5189 8H5.49987C5.24499 7.99972 4.99984 7.90212 4.8145 7.72715C4.62916 7.55218 4.51763 7.31305 4.5027 7.05861C4.48776 6.80416 4.57054 6.55362 4.73413 6.35817C4.89772 6.16271 5.12977 6.0371 5.38287 6.007L5.49987 6ZM10.3829 16.007L10.4999 16H14.4999C14.7547 16.0003 14.9999 16.0979 15.1852 16.2728C15.3706 16.4478 15.4821 16.687 15.497 16.9414C15.512 17.1958 15.4292 17.4464 15.2656 17.6418C15.102 17.8373 14.87 17.9629 14.6169 17.993L14.4999 18H10.4999C10.245 17.9997 9.99984 17.9021 9.8145 17.7272C9.62916 17.5522 9.51763 17.313 9.5027 17.0586C9.48776 16.8042 9.57054 16.5536 9.73413 16.3582C9.89772 16.1627 10.1298 16.0371 10.3829 16.007Z"
                      fill="white"
                    />
                  </svg>
                  <span>
                    Filter
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>,
]
`;
